lib/extended-css-parser.js

```diff
@@ -77,130 +77,131 @@ var ExtendedCssParser = (function() { // jshint ignore:line
+                    this.error(nextIndex);
+                }
 
-            if (lastTokens[l - 1].type === ' ') {
-                lastTokens.length = l - 1;
-            }
-            selector = selector.trim();
-
-            // Start of style declaration parse loop 
-            var styles = Object.create(null);
-            var match, matchPos, matched;
-            nextIndex++; // Move the pointer to the start of style declaration.
-            while (true) {
-                untilClosingBracket: {
-                    // Expects ":", ";", and "}".
-                    reDeclDivider.lastIndex = nextIndex;
-                    match = reDeclDivider.exec(cssText);
-                    if (match === null) {
-                        parseError(nextIndex);
-                    }
-                    matchPos = match.index;
-                    matched = match[0];
-                    if (matched === '}') { break untilClosingBracket; }
-                    if (matched === ':') {
-                        var colonIndex = matchPos;
-                        // Expects ";" and "}".
-                        reDeclEnd.lastIndex = colonIndex;
-                        match = reDeclEnd.exec(cssText);
-                        if (match === null) {
-                            parseError(colonIndex);
-                        }
-                        matchPos = match.index;
-                        matched = match[0];
-                        // Populates the `styles` key-value map.
-                        var property = cssText.slice(nextIndex, colonIndex).trim();
-                        var value = cssText.slice(colonIndex + 1, matchPos).trim();
-                        styles[property] = value;
-                        // If found "}", re-run the outer loop.
-                        if (matched === '}') { break untilClosingBracket; }
-                    }
-                    // matchPos is the position of the next ';'.
-                    // Increase 'nextIndex' and re-run the loop.
-                    nextIndex = matchPos + 1;
-                    continue;
+                // Selector may end with whitespaces, and Sizzle
+                // treat it as a descendant combinator.
+                // We strip it out.
+                var lastTokens = tokens[tokens.length - 1];
+                var l = lastTokens.length;
+
+                if (lastTokens[l - 1].type === ' ') {
+                    lastTokens.length = l - 1;
                 }
+                selector = selector.trim();
+
+                this.nextIndex++; // Move the pointer to the start of style declaration.
 
-                // Matched a closing bracket - cut out matched portion
-                // from cssText and re-run the outer loop.
-                reNonWhitespace.lastIndex = matchPos + 1;
-                match = reNonWhitespace.exec(cssText);
+                var styleMap = this.parseNextStyle();
+
+                results.push({
```

Comment:
Empty styleMap is not handled

Reply:
Worth a unit-test mbe?

Reply:
> Empty styleMap is not handled

Reply:
> Do we need any special treatment?

Nvm, I thought that empty style signals about a mistake. I can see that it's not the case actually.